FRAuthenticator/FRAuthenticator/Policy/FRAPolicyEvaluator.swift

```diff
@@ -0,0 +1,152 @@
+        }
+    }
+    
+    /// Evaluate all registered Policies against an URI.
+    /// - Parameter uri: URL of QR Code
+    public func evaluate(uri: URL) -> Result {
+        let policiesJson = getPoliciesFromURI(uri: uri)
+        return processPolicies(policiesJson: policiesJson)
+    }
+ 
+    /// Evaluate all registered Policies against an URI.
+    /// - Parameter account: Account object
+    public func evaluate(account: Account) -> Result {
+        return processPolicies(policiesJson: account.policies)
+    }
+    
+    /// Return if a policy was attached to the `Account`.
+    /// - Parameters:
+    ///   - account: Account object
+    ///   - policyName: The name of the policy
+    /// - Returns: true, if the policy was attached to the Account, false otherwise.
+    public func isPolicyAttached(account: Account, policyName: String) -> Bool {
+        if let policiesDictionary = FRJSONEncoder.jsonStringToDictionary(jsonString: account.policies!) {
+            return policiesDictionary.keys.contains(policyName)
+        } else {
+            return false;
+        }
+    }
+    
+    
+    //  MARK: - Private
+    
+    private func processPolicies(policiesJson: String?) -> Result {
+        if (policiesJson != nil) {
+            self.targetedPolicies = getPoliciesToVerify(policiesJson: policiesJson!)
+            
+            for policy in self.targetedPolicies {
+                if(!policy.evaluate()) {
+                    return Result(comply: false, nonCompliancePolicy: policy)
+                }
+            }
+        }
+
+        return Result(comply: true, nonCompliancePolicy: nil)
+    }
+    
+    private func getPoliciesToVerify(policiesJson: String) -> [FRAPolicy] {
+        var targetedPolicies = [FRAPolicy]()
+
+        if let policiesDictionary = FRJSONEncoder.jsonStringToDictionary(jsonString: policiesJson) {
+            for policy in self.policies! {
+                if let data = policiesDictionary[policy.name] {
+                    policy.data = data
+                    targetedPolicies.append(policy)
+                }
+            }
+        } else {
+            FRALog.v("No policies to be verified")
+        }
+        
+        return targetedPolicies
+    }
+    
+    private func getPoliciesFromURI(uri: URL) -> String? {
+        if let components = URLComponents(url: uri, resolvingAgainstBaseURL: false), let queryItems = components.queryItems {
+            var params: [String: String] = [:]
+            for item in queryItems {
+                if let strVal = item.value {
+                    params[item.name] = strVal
+                }
+            }
+            
+            if let policies = params["policies"], let pUrlData = policies.decodeURL(), let pUrlStr = String(data: pUrlData, encoding: .utf8) {
+                return pUrlStr
+            } else {
+                return nil
+            }
+            
+        } else {
+            return nil
+        }
+    }
+    
+}
+
+
```

Comment:
I see a few force unwrapping cases in this class. I would suggest to avoid using them even if we are currently sure that they have values. Better be safe if later something changes.

Reply:
agree